test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionMetadataProviderTest.cs

```diff
@@ -148,6 +148,29 @@ public void FromStringExpression_SetsContainerAsExpected()
             Assert.Same(myModel, metadata.Container.Model);
         }
 
+        // This is a special case that's basically only possible with Razor Pages.
```

Comment:
Why is the `private` property case only possible with Razor Pages?

More generally, tag and HTML helpers would see similar NREs with `internal` properties that are visible in the application assembly but not to MVC, `static` properties, and indexers. &hellip; Speaking of which, please add tests abusing `static` properties and indexers.

Might also hit issues with `public` fields and perhaps method groups.